### ISSUE TITLE
Popup window not working when file is short - vim error E315

### ISSUE THREAD

**Issue by iddinev** [junegunn/fzf.vim#980]
- [x] I have read through https://github.com/junegunn/fzf.vim/blob/master/README.md
- [x] I have read through https://github.com/junegunn/fzf/blob/master/README-VIM.md
- [x] I have read through the manual page of fzf (`man fzf`)
- [x] I have searched through the existing issues

I know that the title is garbage, but I don't how to word it better, because this absolutely is the problem. I am not even sure it has to do with FZF, but I have no idea how to continue debugging.

I have the :Buffers command set up to use a popup window:
```
command! -bar -bang -nargs=? -complete=buffer FZFBuffers
\ call fzf#vim#buffers(<q-args>,
\     fzf#vim#with_preview({'options': ['--height=100%', '--bind=f1:abort'],  
\         'window':{'width': 0.3, 'height': 0.3}}, 'down:50%'), <bang>0)
```

When I have a file opened that is just a couple of lines long, and I try to open the buffer list I'm greeted with the following:
![image](https://user-images.githubusercontent.com/11410938/78993248-f618c600-7b45-11ea-825d-c6cfd93051e4.png)
After I click 'Enter' I get the popup window on top of my screen and I can't remove it, when I issue the :Buffers command it gets filled up with the relevant info and I can clear it, but the border always stays:
![image](https://user-images.githubusercontent.com/11410938/78993340-2f513600-7b46-11ea-9788-6315e663eef0.png)

I found the issue is linked to the amount of file lines showed on the screen and the sizing settings of the popup window. If the popup window is short enough (y wise) and/or near the top, you don't need to have many lines showing in order not to have the problem.

Hope my ramblings make sense - any help and/or suggestions how to continue debugging are appreciated.

**Comment by junegunn** [comment]
It looks like an issue of your Vim/Neovim installation. Try upgrading it to the latest.

**Comment by iddinev** [comment]
I'm using Vim, and I just double checked - I'm using the latest version available for my OS.
Any suggestion how I can continue debugging this ?

**Comment by jebby** [comment]
I came across this issue earlier today and decided to toy around and see if I could find what was causing the problem.

In my case, this happens when the value of `scrolloff` is greater than `winheight()` and the popup window has not been opened yet.

It seems that changing the value `scrolloff` after initially opening the popup does not cause the error

**Comment by tuna-f1sh** [comment]
I'm also getting this error, vim: stable 8.2.1700, FZF 0.22.0. 

Above from @jebby helped find a solution: I have `scrolloff` set to 9999 in my vimrc to centre the cursor at all times. Removing this for default value prevents the error. I'd prefer to keep cursor centre however...

**Comment by iddinev** [comment]
> I came across this issue earlier today and decided to toy around and see if I could find what was causing the problem.
> 
> In my case, this happens when the value of `scrolloff` is greater than `winheight()` and the popup window has not been opened yet.
> 
> It seems that changing the value of `scrolloff` after initially opening the popup does not cause the error

This starts to seem a bit like a vim bug, I'll try to reproduce it with an empty, manually created popup window.

**Comment by arekmd** [comment]
I had the same issue. 
Removing  "let so=999" fixed it.

Interestingly, :Files command ended up with error, and empty window, but command :FZF worked fine.

**Comment by iddinev** [comment]
Issue seems fixed now (fzf 0.25.1 (f55c990), vim 8.2 patches 1-2489) dunno which project fixed it though.
